// .ignored/@sveltepress/theme-default/dist/components/layout.js
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
import { get, writable } from 'svelte/store';
import themeOptions from 'virtual:sveltepress/theme-default';
export var MOBILE_EDGE_WIDTH = 950;
export var sidebarCollapsed = writable(true);
export var tocCollapsed = writable(true);
export var navCollapsed = writable(true);
export var anchors = writable([]);
export var pages = writable([]);
export var scrollY = writable(0);
export var oldScrollY = writable(0);
export var scrollDirection = writable('up');
export var darkMode = writable('auto');
export var isDark = writable(false);
export var sidebar = writable(true);
export var showHeader = writable(true);
export var showLayout = writable(true);
export var resolvedSidebar = writable(Object.entries((themeOptions.sidebar || {})).reduce(function (all, _a) {
    var item = _a[1];
    return __spreadArray(__spreadArray([], all, true), item, true);
}, []));
function flattenPages(items) {
    var result = [];
    for (var _i = 0, items_1 = items; _i < items_1.length; _i++) {
        var item = items_1[_i];
        if (item.to)
            result.push(item);
        if (Array.isArray(item.items))
            result.push.apply(result, flattenPages(item.items));
    }
    return result;
}
scrollY.subscribe(function (sy) {
    var nextDirection = sy - get(oldScrollY) > 0 ? 'down' : 'up';
    if (nextDirection !== get(scrollDirection))
        scrollDirection.set(nextDirection);
});
resolvedSidebar.subscribe(function (sidebar) {
    pages.set(flattenPages(sidebar));
});
sidebarCollapsed.subscribe(function (v) {
    if (!v)
        tocCollapsed.set(true);
});
tocCollapsed.subscribe(function (v) {
    if (!v)
        sidebarCollapsed.set(true);
});
export function resolveSidebar(routeId) {
    var _a;
    if (!routeId)
        return;
    var normalizedRouteId = routeId.replace(/\/$/, '');
    var key = Object.keys(themeOptions.sidebar || {}).find(function (key) {
        return normalizedRouteId.startsWith(key.replace(/\/$/, ''));
    });
    // If no matching key found, clear the sidebar
    if (!key) {
        resolvedSidebar.set([]);
        return;
    }
    resolvedSidebar.set(((_a = themeOptions.sidebar) === null || _a === void 0 ? void 0 : _a[key]) || []);
}
