// @sveltepress/theme-default@7.3.2/dist/auto-sidebar.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
import { readdirSync, readFileSync, statSync } from 'node:fs';
import { join, relative, resolve } from 'node:path';
function extractFrontmatter(filePath) {
    try {
        var content = readFileSync(filePath, 'utf-8');
        var match = content.match(/^---\r?\n([\s\S]*?)\r?\n---/);
        if (!match)
            return {};
        var fm = {};
        for (var _i = 0, _a = match[1].split('\n'); _i < _a.length; _i++) {
            var line = _a[_i];
            var colonIdx = line.indexOf(':');
            if (colonIdx === -1)
                continue;
            var key = line.slice(0, colonIdx).trim();
            var value = line.slice(colonIdx + 1).trim();
            // Parse booleans and numbers
            if (value === 'true')
                value = true;
            else if (value === 'false')
                value = false;
            else if (/^-?\d+(?:\.\d+)?$/.test(value))
                value = Number(value);
            fm[key] = value;
        }
        return fm;
    }
    catch (_b) {
        return {};
    }
}
function getPageMeta(filePath) {
    var fm = extractFrontmatter(filePath);
    return {
        title: fm.sidebarTitle || fm.title || '',
        order: typeof fm.order === 'number' ? fm.order : 100,
        sidebar: fm.sidebar !== false,
        sidebarTitle: fm.sidebarTitle,
        collapsible: fm.collapsible,
    };
}
function humanize(str) {
    return str
        .replace(/[-_]/g, ' ')
        .replace(/\b\w/g, function (c) { return c.toUpperCase(); });
}
function isGroupDir(name) {
    return /^\(.*\)$/.test(name);
}
function isDynamicRoute(name) {
    return /^\[.*\]$/.test(name);
}
function isPrivate(name) {
    return name.startsWith('_');
}
function shouldSkipDir(name) {
    return isDynamicRoute(name) || isPrivate(name);
}
function findPageFile(dirPath) {
    for (var _i = 0, _a = ['+page.md', '+page.svelte']; _i < _a.length; _i++) {
        var name = _a[_i];
        var filePath = join(dirPath, name);
        try {
            if (statSync(filePath).isFile())
                return filePath;
        }
        catch (_b) { }
    }
    return null;
}
function getSubDirs(dirPath) {
    try {
        return readdirSync(dirPath).filter(function (name) {
            if (name.startsWith('+') || name.startsWith('.'))
                return false;
            try {
                return statSync(join(dirPath, name)).isDirectory();
            }
            catch (_a) {
                return false;
            }
        });
    }
    catch (_a) {
        return [];
    }
}
/**
 * Build the URL path from a filesystem path relative to routes dir.
 * Strips parenthesized group segments like (group).
 */
function buildUrlPath(relativePath) {
    var segments = relativePath.split(/[/\\]/).filter(Boolean);
    var urlSegments = segments.filter(function (s) { return !isGroupDir(s); });
    return "/".concat(urlSegments.join('/'), "/");
}
/**
 * Recursively scan a directory and build sidebar items.
 * @param dirPath - absolute path to the current directory
 * @param routesDir - absolute path to the routes root
 * @param depth - current nesting depth (0 = root of a sidebar section)
 */
function scanDir(dirPath, routesDir, depth) {
    var subDirs = getSubDirs(dirPath);
    var items = [];
    var _loop_1 = function (name) {
        if (shouldSkipDir(name))
            return "continue";
        var subDirPath = join(dirPath, name);
        // Handle group directories — they are transparent in URL structure
        if (isGroupDir(name)) {
            var groupItems = scanDir(subDirPath, routesDir, depth);
            items.push.apply(items, groupItems.map(function (item) { var _a; return (__assign(__assign({}, item), { _order: (_a = item._order) !== null && _a !== void 0 ? _a : 100 })); }));
            return "continue";
        }
        var pageFile = findPageFile(subDirPath);
        var childDirs = getSubDirs(subDirPath).filter(function (n) { return !shouldSkipDir(n) && !isPrivate(n); });
        // Also check group dirs for children
        var hasGroupChildren = childDirs.some(function (n) { return isGroupDir(n) && getSubDirs(join(subDirPath, n)).length > 0; });
        var hasRealChildren = childDirs.some(function (n) { return !isGroupDir(n); });
        var hasChildren = hasRealChildren || hasGroupChildren;
        if (!pageFile && !hasChildren)
            return "continue";
        var meta = pageFile ? getPageMeta(pageFile) : { title: '', order: 100, sidebar: true, collapsible: undefined };
        if (!meta.sidebar)
            return "continue";
        var title = meta.title || humanize(name);
        var relPath = relative(routesDir, subDirPath);
        var urlPath = buildUrlPath(relPath);
        if (hasChildren) {
            // This is a group with nested items
            var nestedItems = scanDir(subDirPath, routesDir, depth + 1);
            var item = {
                title: title,
                _order: meta.order,
                items: nestedItems,
            };
            if (meta.collapsible !== undefined)
                item.collapsible = meta.collapsible;
            // If this dir also has its own page, add it as first item
            if (pageFile) {
                var selfItem = {
                    title: title,
                    to: urlPath,
                };
                item.items = __spreadArray([selfItem], nestedItems, true);
            }
            items.push(item);
        }
        else {
            // Leaf page
            items.push({
                title: title,
                to: urlPath,
                _order: meta.order,
            });
        }
    };
    for (var _i = 0, subDirs_1 = subDirs; _i < subDirs_1.length; _i++) {
        var name = subDirs_1[_i];
        _loop_1(name);
    }
    // Sort by order, then alphabetically
    items.sort(function (a, b) {
        if (a._order !== b._order)
            return a._order - b._order;
        return (a.title || '').localeCompare(b.title || '');
    });
    // Strip _order from output
    return items.map(function (_a) {
        var _order = _a._order, rest = __rest(_a, ["_order"]);
        return rest;
    });
}
/**
 * Detect root sidebar paths from the routes directory.
 * Returns paths like ['/guide/', '/reference/'].
 */
function detectRoots(routesDir) {
    var dirs = getSubDirs(routesDir);
    var roots = [];
    for (var _i = 0, dirs_1 = dirs; _i < dirs_1.length; _i++) {
        var name = dirs_1[_i];
        if (shouldSkipDir(name) || isPrivate(name))
            continue;
        var dirPath = join(routesDir, name);
        if (isGroupDir(name)) {
            // Look inside group dirs for real route dirs
            var innerDirs = getSubDirs(dirPath);
            for (var _a = 0, innerDirs_1 = innerDirs; _a < innerDirs_1.length; _a++) {
                var inner = innerDirs_1[_a];
                if (!shouldSkipDir(inner) && !isPrivate(inner) && !isGroupDir(inner))
                    roots.push("/".concat(inner, "/"));
            }
            continue;
        }
        // Only include dirs that have at least one page somewhere inside
        var hasPages = hasPageFiles(dirPath);
        if (hasPages)
            roots.push("/".concat(name, "/"));
    }
    return roots;
}
function hasPageFiles(dirPath) {
    if (findPageFile(dirPath))
        return true;
    for (var _i = 0, _a = getSubDirs(dirPath); _i < _a.length; _i++) {
        var name = _a[_i];
        if (hasPageFiles(join(dirPath, name)))
            return true;
    }
    return false;
}
/**
 * Find the actual filesystem dir for a root path, handling group dirs.
 */
function findRootDir(routesDir, rootPath) {
    // rootPath like '/guide/' -> segment 'guide'
    var segment = rootPath.replace(/^\/|\/$/g, '').split('/')[0];
    if (!segment)
        return null;
    // Direct match
    var directPath = join(routesDir, segment);
    try {
        if (statSync(directPath).isDirectory())
            return directPath;
    }
    catch (_a) { }
    // Check inside group dirs
    for (var _i = 0, _b = getSubDirs(routesDir); _i < _b.length; _i++) {
        var name = _b[_i];
        if (isGroupDir(name)) {
            var groupPath = join(routesDir, name, segment);
            try {
                if (statSync(groupPath).isDirectory())
                    return groupPath;
            }
            catch (_c) { }
        }
    }
    return null;
}
export function isAutoSidebarOptions(sidebar) {
    return typeof sidebar === 'object'
        && sidebar !== null
        && 'enabled' in sidebar
        && sidebar.enabled === true;
}
/**
 * Generate sidebar configuration by scanning the routes directory.
 */
export function generateSidebar(autoOptions) {
    var routesDir = resolve(autoOptions.routesDir || 'src/routes');
    var roots = autoOptions.roots || detectRoots(routesDir);
    var sidebar = {};
    for (var _i = 0, roots_1 = roots; _i < roots_1.length; _i++) {
        var root = roots_1[_i];
        var rootDir = findRootDir(routesDir, root);
        if (!rootDir)
            continue;
        var items = scanDir(rootDir, routesDir, 0);
        if (items.length > 0)
            sidebar[root] = items;
    }
    return sidebar;
}
