// .ignored/prettier-plugin-markdown-list-tabwidth/index-full-replace.mjs
import * as prettier333MarkdownPlugin from "./prettier-3.3.3-markdown-plugin.mjs";

// Exports the prettier v3.3.3 Markdown language.
export const languages = [prettier333MarkdownPlugin.languages.find((language) => language.name === "Markdown")];
// Exports the prettier v3.3.3 Markdown options.
export const options = prettier333MarkdownPlugin.options;
export const parsers = {
    markdown: {
        // Replaces the prettier Markdown parser with the one from Prettier v3.3.3.
        ...prettier333MarkdownPlugin.parsers.markdown,
        // Overrides the prettier Markdown parser's AST format.
        astFormat: "prettier-3.3.3-mdast",
    },
    mdx: {
        // Replaces the prettier MDX parser with the one from Prettier v3.3.3.
        ...prettier333MarkdownPlugin.parsers.mdx,
        // Overrides the prettier MDX parser's AST format.
        astFormat: "prettier-3.3.3-mdast",
    },
    remark: {
        // Replaces the prettier Remark parser with the one from Prettier v3.3.3.
        ...prettier333MarkdownPlugin.parsers.remark,
        // Overrides the prettier Remark parser's AST format.
        astFormat: "prettier-3.3.3-mdast",
    },
};
export const printers = {
    // Adds the prettier v3.3.3 Markdown printer.
    "prettier-3.3.3-mdast": prettier333MarkdownPlugin.printers.mdast,
};

export default {
    languages,
    options,
    parsers,
    printers,
};
