// prettier-plugin-markdown-list-tabwidth@1.2.0/index.mjs
import * as prettier333MarkdownPlugin from "./prettier-3.3.3-markdown-plugin.mjs";

export const parsers = {
    markdown: {
        // The prettier Markdown parser from the currently installed prettier version.
        ...prettier333MarkdownPlugin.parsers.markdown,
        // Overrides the prettier Markdown parser's AST format.
        astFormat: "prettier-3.3.3-mdast",
    },
    mdx: {
        // The prettier MDX parser from the currently installed prettier version.
        ...prettier333MarkdownPlugin.parsers.mdx,
        // Overrides the prettier MDX parser's AST format.
        astFormat: "prettier-3.3.3-mdast",
    },
    remark: {
        // The prettier Remark parser from the currently installed prettier version.
        ...prettier333MarkdownPlugin.parsers.remark,
        // Overrides the prettier Remark parser's AST format.
        astFormat: "prettier-3.3.3-mdast",
    },
};
export const printers = {
    // Adds the prettier v3.3.3 Markdown printer.
    "prettier-3.3.3-mdast": prettier333MarkdownPlugin.printers.mdast,
};

export default {
    parsers,
    printers,
};
